opentech/apply/projects/views.py

```diff
@@ -11,20 +12,81 @@
 from opentech.apply.activity.messaging import MESSAGES, messenger
 from opentech.apply.activity.views import ActivityContextMixin, CommentFormView
 from opentech.apply.users.decorators import staff_required
-from opentech.apply.utils.views import (DelegateableView, DelegatedViewMixin,
-                                        ViewDispatcher)
+from opentech.apply.utils.views import (
+    DelegateableView,
+    DelegatedViewMixin,
+    ViewDispatcher
+)
 
 from .forms import (
+    ApproveContractForm,
     CreateApprovalForm,
     ProjectApprovalForm,
     ProjectEditForm,
     RejectionForm,
     RemoveDocumentForm,
     SetPendingForm,
     UpdateProjectLeadForm,
-    UploadDocumentForm,
+    UploadContractForm,
+    UploadDocumentForm
 )
-from .models import CONTRACTING, Approval, Project, PacketFile
+from .models import CONTRACTING, Approval, Contract, PacketFile, Project
+
+
+class ContractsMixin:
+    def get_context_data(self, **kwargs):
+        project = self.get_object()
+        recent_contracts = list(project.contracts.filter(is_signed=True)
+                                                 .order_by('-created_at')[:5])
```

Comment:
not sure the limit is helpful without show more functionality, it would just cause confusion in the future..

Reply:
For some reason I thought we wanted this limited… not sure where I got that from!

Fixed and rebased.